internal/pkg/sourcetype/sourcetype.go

```diff
@@ -30,28 +30,36 @@ import (
 // - A Struct Type that contains a tagged field
 // - A composite type that contains a Source Type
 func IsSourceType(c *config.Config, tf fieldtags.ResultType, t types.Type) bool {
-	deref := utils.Dereference(t)
-	switch tt := deref.(type) {
+	seen := map[types.Type]bool{}
+	return isSourceType(c, tf, t, seen)
+}
+
+func isSourceType(c *config.Config, tf fieldtags.ResultType, t types.Type, seen map[types.Type]bool) bool {
+	if seen[t] {
```

Comment:
would be nice to explain why if its see then it can't be source type.

Reply:
Done. Also added a comment for the function itself. Let me know what you think.